apps/ledger-live-desktop/src/renderer/screens/exchange/Swap2/Form/index.tsx

```diff
@@ -291,9 +292,13 @@ const SwapForm = () => {
       const { feesStrategy } = transaction || {};
       const { rate, rateId } = exchangeRate || {};
 
+      const isToAccountValid = totalListedAccounts.some(account => account.id === toAccount?.id);
       const fromAccountId =
         fromAccount && accountToWalletAPIAccount(fromAccount, fromParentAccount)?.id;
-      const toAccountId = toAccount && accountToWalletAPIAccount(toAccount, toParentAccount)?.id;
+      const toAccountId = isToAccountValid
+        ? toAccount && accountToWalletAPIAccount(toAccount, toParentAccount)?.id
+        : toParentAccount && accountToWalletAPIAccount(toParentAccount, null)?.id;
+      const toNewTokenId = !isToAccountValid ? toAccount?.token?.id : null;
```

Comment:
We should be able to remove the `isToAccountValid` and `totalListedAccounts` variables by checking the account itself.

But this change means that we send the parent account for all tokens and not only new ones, it's something to take into account if you want to use my suggestion.

Also we can be safe and just go with your solution for now this is only a suggestion.

```suggestion
      const toAccountId = isTokenAccount(toAccount)
        ? toParentAccount && accountToWalletAPIAccount(toParentAccount, null)?.id
        : toAccount && accountToWalletAPIAccount(toAccount, toParentAccount)?.id;
      const toNewTokenId = isTokenAccount(toAccount) ? toAccount.token.id : undefined;
```